app/controllers/base_published_record_controller.rb

```diff
@@ -93,10 +96,13 @@ def revisions
   end
 
   def revert
-    latest_revision_id = @revisions.first['meta']['revision-id']
+    latest_revision_id = @revisions.dig(0, 'meta', 'revision-id')
+
+    # grab the collection_association of the correct revision
+    collection_association = @revisions.dig(@revision_id.to_i, 'associations', 'collections', 0, 'concept-id')
```

Comment:
This isn't grabbing the collection association. Please make the name more descriptive of what it is.